### ISSUE TITLE
Celerite cannot be installed with pip, pip3 and from source (PYBIND11_PLUGIN deprecated) 

### ISSUE THREAD

**Issue by 3fon3fonov** [dfm/celerite#136]
First of all thank you very much for Celerite!

I have it working on several machines, which I installed a couple of  months ago.

Now I tried to install the package on a Virtual machine with Ubuntu 16.04, 18.04 and Lubuntu 18.04 
and the installation did not went as expected.
During the installation the OS becomes almost unresponsive (severe lag). 

Installation was tried with pip, pip3 and install from the git clone. 

What I remember from the git-clone source install was a warning like 
"PYBIND11_PLUGIN deprecated use PYBIND11_MODULE instead." 

(Sorry I cannot copy the error because My OS blocks at the terminal)

I have pybind 2.2.4

Thanks,
Trifon

**Comment by dfm** [comment]
Strange - I don't think it can be caused by the deprecation warning that you saw because I think that that version of pybind11 still implements the old interface. I expect this is somehow a memory issue. The compilation step is pretty memory intensive for various reasons (Eigen, autodiff, etc.) so my guess is that your virtual machine is running out of RAM but I'm definitely not 100% on that. Let me know if you can play around with that a bit and I'll see if I can reproduce. A temporary option might be to use the version on conda-forge...

**Comment by 3fon3fonov** [comment]
Hi Dan, 

Thanks for the fast replay. As far as I remember my Virt.box. OS systems are all running on 2GB of RAM.  This indeed may be the problem, so let me try again with say 4 GB? I will report soon.

**Comment by 3fon3fonov** [comment]
Dear Dan,

I can confirm your suspicion! When I increased the OS RAM to 4 GB the installation went with no problem. Meanwhile I wan monitoring the memory usage. cc1plus takes nearly 2.5 GB! I would have never guessed that installing celerite (1.3 Mb) would generate such a memory overhead during installation. With this the problem is resolved and I am closing this issue. 

Thanks,
Trifon
